modules/packages/LinearAlgebra.chpl

```diff
@@ -221,6 +222,260 @@ class LinearAlgebraError : Error {
+      isAComputed[5] = true;
+    }
+    return this.A10;
+  }
+
+  proc matrix_prod(A: [], B: []){
+    return dot(A,B);
+  }
+
+  proc comp_Exactd4(){
+    if !isExactdComputed[1] then {
+      this.Exactd4 = norm(comp_A4(), normType.norm1)**(1/4.0);
+      isExactdComputed[1] = true;
+    }
+    return this.Exactd4;
+  }
+
+  proc comp_Exactd6(){
+    if !isExactdComputed[2] then {
+      this.Exactd6 = norm(comp_A6(), normType.norm1)**(1/6.0);
+      isExactdComputed[2] = true;
+    }
+    return this.Exactd6;
+  }
+
+  proc comp_Exactd8(){
+    if !isExactdComputed[3] then {
+      this.Exactd8 = norm(comp_A8(), normType.norm1)**(1/8.0);
+      isExactdComputed[3] = true;
+    }
+    return this.Exactd8;
+  }
+
+  proc comp_Exactd10(){
+    if !isExactdComputed[4] then {
+      this.Exactd10 = norm(comp_A10(), normType.norm1)**(1/10.0);
+      isExactdComputed[4] = true;
+    }
+    return this.Exactd10;
+  }
+
+  proc comp_Loosed4(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd4();
+    }
+    else {
+      if !isApproxdComputed[1] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd4 = norm(comp_A4(), normType.norm1)**(1/4.0);
+        isApproxdComputed[1] = true;
+      }
+      return this.Approxd4;
+    }
+  }
+
+  proc comp_Loosed6(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd6();
+    }
+    else {
+      if !isApproxdComputed[2] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd6 = norm(comp_A6(), normType.norm1)**(1/6.0);
+        isApproxdComputed[2] = true;
+      }
+      return this.Approxd6;
+    }
+  }
+
+  proc comp_Loosed8(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd8();
+    }
+    else {
+      if !isApproxdComputed[3] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd8 = norm(comp_A8(), normType.norm1)**(1/8.0);
+        isApproxdComputed[3] = true;
+      }
+      return this.Approxd8;
+    }
+  }
+
+  proc comp_Loosed10(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd10();
+    }
+    else {
+      if !isApproxdComputed[4] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd10 = norm(comp_A10(), normType.norm1)**(1/10.0);
+        isApproxdComputed[4] = true;
+      }
+      return this.Approxd10;
+    }
+  }
+
+  proc pade3(){
+    var b = [120.0, 60.0, 12.0, 1.0];
+    var U = matrix_prod(this.A, reshape(b[3]*comp_A2() + b[1]*this.ident, this.A.indices));
```

Comment:
On this line (and others below), do we need the reshapes? There is probably some cost to each of these.

Reply:
Actually, I forgot to file a bug. There is an issue with the reshape function. Let's say A is a 2*2 matrix then 2*A should also be a 2*2 matrix. But I see that somehow the resultant matrix is getting flattened. Hence for now, I had to use the reshape function.